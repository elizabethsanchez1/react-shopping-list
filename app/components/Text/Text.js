import React from 'react';
import PropTypes from 'prop-types';
import { Text as BaseText, StyleSheet } from 'react-native';
import theme from "../../../app/styles/theme.style";


const styles = StyleSheet.create({
  header: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
  standardText: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
  small: {
    fontSize: theme.FONT_SIZE_SMALL,
  },
  medium: {
    fontSize: theme.FONT_SIZE_MEDIUM,
  },
  large: {
    fontSize: theme.FONT_SIZE_LARGE,
  },
  lightFontWeight: {
    fontWeight: theme.FONT_WEIGHT_LIGHT,
  },
  mediumFontWeight: {
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
  },
  boldFontWeight: {
    fontWeight: theme.FONT_WEIGHT_BOLD,
  }
});


// TODO: Font weight does not work for some reason???

const Text = ({ children, size, weight, textStyling }) => {
  let styleObject = { ...styles.standardText  };

  if ( size === 'small' ) {
    styleObject = { ...styleObject, ...styles.small };
  }

  if ( size === 'medium' ) {
    styleObject = { ...styleObject, ...styles.medium };
  }

  if ( size === 'large' ) {
    styleObject = { ...styleObject, ...styles.large };
  }

  if ( size === 'header' ) {
    styleObject = { ...styleObject, ...styles.header };
  }

  // if ( weight === 'light' ) {
  //   styleObject = { ...styleObject, ...styles.lightFontWeight }
  // }
  //
  // if ( weight === 'medium' ) {
  //   styleObject = { ...styleObject, ...styles.mediumFontWeight }
  // }
  //
  // if ( weight === 'bold' ) {
  //   styleObject = { ...styleObject, ...styles.boldFontWeight }
  // }

  styleObject = { ...styleObject, ...textStyling };

  return (
    <BaseText
      style={ styleObject }
    >
      { children }
    </BaseText>
  )
};

Text.propTypes = {
  children: PropTypes.any,
  size: PropTypes.string,
  textStyling: PropTypes.object,
};

export default Text;
