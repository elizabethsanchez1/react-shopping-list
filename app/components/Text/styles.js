import { create } from 'react-native-platform-stylesheet';
import theme from '../../styles/theme.style';

export default create({
  header: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },

  // Text specific styles

  standardText: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
});
