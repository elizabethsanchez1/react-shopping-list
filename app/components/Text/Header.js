import PropTypes from 'prop-types';
import React from 'react';
import { Text } from 'react-native';
import styles from './styles';

const Header = (props) => (
  <Text
    style={[styles.header, props.style]}
  >
    {props.children}
  </Text>
);

Header.propTypes = {
  children: PropTypes.any,
  style: PropTypes.any,
};

export default Header;
