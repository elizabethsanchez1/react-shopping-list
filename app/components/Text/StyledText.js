import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';
import styles from './styles';

const StyledText = (props) => (
  <Text
    style={[styles.standardText, props.style]}
  >
    {props.children}
  </Text>
);


StyledText.propTypes = {
  children: PropTypes.any,
  style: PropTypes.any,
};

export default StyledText;
