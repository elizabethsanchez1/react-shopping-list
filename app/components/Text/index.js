/* eslint-disable import/prefer-default-export */
import Header from './Header';
import StyledText from './StyledText';
import Text from './Text';

export { Header, StyledText, Text };
