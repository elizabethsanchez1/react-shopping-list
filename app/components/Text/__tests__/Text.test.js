import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from "../../Text";


describe( 'Text Component', () => {

  it( 'Text is visually looks the same', () => {
    const tree = renderer.create(
      <Text>Testing</Text>
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );

} );




