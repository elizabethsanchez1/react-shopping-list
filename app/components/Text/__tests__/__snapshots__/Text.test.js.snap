// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component Text is visually looks the same 1`] = `
<Text
  style={
    Object {
      "color": "#FFF",
      "fontFamily": "RobotoCondensed-Regular",
      "fontSize": 16,
      "fontWeight": "600",
    }
  }
>
  Testing
</Text>
`;
